### PR TITLE
feat(stylus-verifier): make use of docker api

### PR DESCRIPTION
**Pull request by rimrakhimov**

### CODE REVIEW

File: stylus-verifier/stylus-verifier-logic/src/docker.rs [hunk 1]
```diff
@@ -74,65 +109,193 @@ async fn create_image(
...
+RUN rustup target add wasm32-unknown-unknown
+RUN rustup component add rust-src --toolchain {toolchain}-x86_64-unknown-linux-gnu
+"
+    );
+
+    let content = build_tar_with_dockerfile(&dockerfile)?;
+    let build_image_options = BuildImageOptions {
+        t: image_name.to_string(),
+        dockerfile: "Dockerfile".to_string(),
+        version: BuilderVersion::BuilderV1,
```
Comment (bragov4ik) [inline stylus-verifier/stylus-verifier-logic/src/docker.rs:126]: Why not `BuildKit`? AFAIR it's more advanced and newer

File: stylus-verifier/stylus-verifier-logic/src/docker.rs [hunk 2]
```diff
@@ -74,65 +109,193 @@ async fn create_image(
...
+
+    let mut image_build_stream =
+        docker.build_image(build_image_options, None, Some(content.into()));
+
+    let mut output = vec![];
+    while let Some(result) = image_build_stream.next().await {
+        match result {
+            Ok(info) => {
+                if let Some(value) = info.stream {
+                    output.push(value)
```
Comment (bragov4ik) [inline stylus-verifier/stylus-verifier-logic/src/docker.rs:143]: not sure what's returned here, but if it's some progress bar, then ig it makes sense to print it in logs?

File: stylus-verifier/stylus-verifier-logic/src/docker.rs [hunk 3]
```diff
@@ -74,65 +109,193 @@ async fn create_image(
...
+    docker
+        .upload_to_container(container_id, Some(options), tar.into())
         .await
-        .context("failed to execute Docker run command")?;
+        .context("uploading tar archive to container")?;
+
+    Ok(())
+}
+
+async fn start_container(docker: &Docker, container_id: &str) -> Result<String, anyhow::Error> {
```
Comment (bragov4ik) [inline stylus-verifier/stylus-verifier-logic/src/docker.rs:225]: nit:
it seems to not only start but run the container till the end (until output stream ends). Maybe `run_container`?

**Review by bragov4ik — APPROVED** [review]
LGTM, we discussed buildkit in a meeting today so let's see if it works now :)

File: stylus-verifier/stylus-verifier-logic/src/docker.rs (hunk 1 above)
Comment (rimrakhimov) [reply stylus-verifier/stylus-verifier-logic/src/docker.rs:126]: image

Specifying BuildKit unfortunately returns some obscure error of "no active session for {session_id}". The error can be checked at https://github.com/blockscout/blockscout-rs/actions/runs/11158425854/job/31014679882?pr=1074

So, decided to leave BuilderV1 for now

File: stylus-verifier/stylus-verifier-logic/src/docker.rs (hunk 3 above)
Comment (rimrakhimov) [reply stylus-verifier/stylus-verifier-logic/src/docker.rs]: Updated in [c5ce76d](https://github.com/blockscout/blockscout-rs/pull/1074/commits/c5ce76da327956834ca5426ac3c8dae0d13686ef)

File: stylus-verifier/stylus-verifier-logic/src/docker.rs (hunk 2 above)
Comment (rimrakhimov) [reply stylus-verifier/stylus-verifier-logic/src/docker.rs:144]: Added tracing logs in [c5ce76d](https://github.com/blockscout/blockscout-rs/pull/1074/commits/c5ce76da327956834ca5426ac3c8dae0d13686ef)

**Pull request merged**